catalog/app/containers/Bucket/PackageRevisions.js

```diff
@@ -328,9 +203,159 @@ function Revision({ bucket, name, id, hash, stats, message, modified, counts })
         height={{ xs: 64, sm: 'auto' }}
         width={sparklineW}
       >
-        {AsyncResult.case(
+        {counts({ sparklineW, sparklineH })}
+      </M.Box>
+    </M.Paper>
+  )
+}
+
+function RevisionSkel() {
+  const t = M.useTheme()
+  const xs = M.useMediaQuery(t.breakpoints.down('xs'))
+  return (
+    <RevisionLayout
+      link={
+        <M.Box height={20} display="flex" alignItems="center">
+          <Skeleton borderRadius="borderRadius" height={16} width={200} />
+        </M.Box>
+      }
+      msg={
+        <>
+          <M.Box height={20} display="flex" alignItems="center">
+            <Skeleton borderRadius="borderRadius" height={16} width="100%" />
+          </M.Box>
+          {!xs && (
+            <M.Box height={20} display="flex" alignItems="center">
+              <Skeleton borderRadius="borderRadius" height={16} width="90%" />
+            </M.Box>
+          )}
+        </>
+      }
+      hash={
+        <Skeleton borderRadius="borderRadius" height={16} width="100%" maxWidth={550} />
+      }
+      stats={
+        <M.Box display="flex" alignItems="center">
+          <Skeleton borderRadius="borderRadius" width={70} height={16} mr={2} />
+          <Skeleton borderRadius="borderRadius" width={70} height={16} />
+        </M.Box>
+      }
+      counts={({ sparklineW, sparklineH }) => (
+        <>
+          <M.Box
+            position="absolute"
+            right={16}
+            top={0}
+            display="flex"
+            alignItems="center"
+            height={20}
+            width={120}
+          >
+            <Skeleton height={16} width="100%" borderRadius="borderRadius" />
+          </M.Box>
+          <SparklineSkel width={sparklineW} height={sparklineH} />
+        </>
+      )}
+    />
+  )
+}
+
+const useRevisionStyles = M.makeStyles((t) => ({
+  mono: {
+    fontFamily: t.typography.monospace.fontFamily,
+  },
+  time: {
+    ...t.typography.body1,
+    fontWeight: t.typography.fontWeightMedium,
+    lineHeight: t.typography.pxToRem(20),
+    whiteSpace: 'nowrap',
+  },
+  msg: {
+    WebkitBoxOrient: 'vertical',
+    WebkitLineClamp: 2,
+    display: '-webkit-box',
```

Comment:
Did you mean `display: 'flex'`?

Reply:
nope, it's a hack (sorta) to only show two lines of text

Reply:
I would consider moving this pattern to the theme's "mixin" and use it like `t.spacing`. For example:

```js
// theme
createMuiTheme({
  trimLines(linesCount) => ({
     WebkitBoxOrient: 'vertical',
     WebkitLineClamp: 2,
     display: '-webkit-box',
  }),
})
```

```js
// component
makeStyles((t) => {
  msg: {
    ...t.trimLines(2),
  },
})
```

Reply:
yeah, good idea. done